scipy/_lib/_array_api.py

```diff
@@ -335,3 +335,7 @@ def cov(x, *, xp=None):
     c /= fact
     axes = tuple(axis for axis, length in enumerate(c.shape) if length == 1)
     return xp.squeeze(c, axis=axes)
+
+
+def xp_unsupported_param_msg(param):
+    return f'Providing {param!r} is only supported for numpy arrays.'
```

Comment:
Seems straightforward enough if you need this more generally as you suggest. I guess we can leave this open for a bit to see if folks might prefer an adjustment to the name, although I'm not sure it is worth too much energy since it should be private/internal to our own testing.